prboom2/src/p_setup.c

```diff
@@ -3307,7 +3307,7 @@ static void R_CalcSegsLength(void)
     int64_t dx = (int64_t)li->v2->px - li->v1->px;
     int64_t dy = (int64_t)li->v2->py - li->v1->py;
     length = sqrt((double)dx*dx + (double)dy*dy);
-    li->length = (int64_t)length;
+    li->length = (uint32_t)(length / 2.0);
```

Comment:
Should we rather convert to 64 bits, shift right, and convert to 32 bits, so it's exactly the same as before?

Reply:
I can't think of a way the results could be different.  Floating point division by 2.0 should be exact for non-subnormal values (subtracting 1 from the exponent), and subnormal inputs should be impossible (the smallest non-zero length will be `sqrt(2)`).  Truncating twice instead of once should also not change the result.